File: apis/fluent/atrium-api-fluent-java/src/test/java/ch/tutteli/atrium/api/fluent/java/ComparableExpectationsSpec.java
package ch.tutteli.atrium.api.fluent.java;

import ch.tutteli.atrium.translations.DescriptionComparableExpectation;

public class ComparableExpectationsSpec extends ch.tutteli.atrium.specs.integration.ComparableExpectationsSpec {
    ComparableExpectationsSpec() {
        super(
            null, //ch.tutteli.atrium.api.fluent.java.ComparableExpectationsSpec.pair("toBeLessThan",  (coreExpect, integer) -> expect(coreExpect).toBeLessThan(integer).getCoreExpect()),
            null,//new Pair<String, Function2<Expect<Integer>, Integer, Expect<Integer>>>("toBeLessThan", (coreExpect, integer) -> expect(coreExpect).toBeLessThan(integer).getCoreExpect()),
            null,//new Pair<String, Function2<Expect<Integer>, Integer, Expect<Integer>>>("toBeLessThan", (coreExpect, integer) -> expect(coreExpect).toBeLessThan(integer).getCoreExpect()),
            null,//new Pair<String, Function2<Expect<Integer>, Integer, Expect<Integer>>>("toBeLessThan", (coreExpect, integer) -> expect(coreExpect).toBeLessThan(integer).getCoreExpect()),
            null,//new Pair<String, Function2<Expect<Integer>, Integer, Expect<Integer>>>("toBeLessThan", (coreExpect, integer) -> expect(coreExpect).toBeLessThan(integer).getCoreExpect()),
           null,//new Pair<String, Function2<Expect<DiffEqualsCompareTo>, DiffEqualsCompareTo, Expect<DiffEqualsCompareTo>>>("toBeLessThan", (coreExpect, integer) ->coreExpect),
           null,//new Pair<String, Function2<Expect<DiffEqualsCompareTo>, DiffEqualsCompareTo, Expect<DiffEqualsCompareTo>>>("toBeLessThan", (coreExpect, integer) ->coreExpect),
           null,//new Pair<String, Function2<Expect<DiffEqualsCompareTo>, DiffEqualsCompareTo, Expect<DiffEqualsCompareTo>>>("toBeLessThan", (coreExpect, integer) ->coreExpect),
            DescriptionComparableExpectation.TO_BE_LESS_THAN_OR_EQUAL_TO.getDefault(),
            DescriptionComparableExpectation.TO_BE_GREATER_THAN_OR_EQUAL_TO.getDefault(),
            "[Atrium]"
        );
    }
}

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/ComparableExpectationsSpec.kt
package ch.tutteli.atrium.api.infix.en_GB

import ch.tutteli.atrium.creating.Expect
import ch.tutteli.atrium.specs.fun1
import ch.tutteli.atrium.specs.integration.DiffEqualsCompareTo
import ch.tutteli.atrium.specs.notImplemented
import ch.tutteli.atrium.translations.DescriptionComparableExpectation
import org.spekframework.spek2.Spek

class ComparableExpectationsSpec: Spek({

    include(object: ch.tutteli.atrium.specs.integration.ComparableExpectationsSpec(
        fun1(Expect<Int>::toBeLessThan),
        fun1(Expect<Int>::toBeLessThanOrEqualTo),
        fun1(Expect<Int>::toBeEqualComparingTo),
        fun1(Expect<Int>::toBeGreaterThanOrEqualTo),
        fun1(Expect<Int>::toBeGreaterThan),

        fun1(Expect<DiffEqualsCompareTo>::toBeLessThanOrEqualTo),
        fun1(Expect<DiffEqualsCompareTo>::toBeEqualComparingTo),
        fun1(Expect<DiffEqualsCompareTo>::toBeGreaterThanOrEqualTo),
        describePrefix = "[Atrium][<=] "
    ){})

    include(object: ch.tutteli.atrium.specs.integration.ComparableExpectationsSpec(
        fun1(Expect<Int>::toBeLessThan),
        fun1(Expect<Int>::notToBeGreaterThan),
        fun1(Expect<Int>::toBeEqualComparingTo),
        fun1(Expect<Int>::notToBeLessThan),
        fun1(Expect<Int>::toBeGreaterThan),

        fun1(Expect<DiffEqualsCompareTo>::notToBeGreaterThan),
        fun1(Expect<DiffEqualsCompareTo>::toBeEqualComparingTo),
        fun1(Expect<DiffEqualsCompareTo>::notToBeLessThan),

        DescriptionComparableExpectation.NOT_TO_BE_GREATER_THAN.getDefault(),
        DescriptionComparableExpectation.NOT_TO_BE_LESS_THAN.getDefault(),
        describePrefix = "[Atrium][!>] "
    ){})
})  {

    @Suppress("unused")
    fun ambiguityTest() {
        val a1: Expect<Int> = notImplemented()
        a1 toBeLessThan 1
        a1 toBeLessThanOrEqualTo 1
        a1 toBeGreaterThan 1
        a1 toBeGreaterThanOrEqualTo 1
        a1 toBeEqualComparingTo 1
    }
}
